src/mono/wasm/runtime/jiterpreter-support.ts

```diff
@@ -0,0 +1,757 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { NativePointer, VoidPtr } from "./types/emscripten";
+import { Module } from "./imports";
+import { WasmOpcode } from "./jiterpreter-opcodes";
+import cwraps from "./cwraps";
+
+// uint16
+export declare interface MintOpcodePtr extends NativePointer {
+    __brand: "MintOpcodePtr"
+}
+
+export class WasmBuilder {
```

Comment:
I love conceptual simplicity of this builder. Great for prototyping.
Few ideas which will spoil it :(

Should we rather emit binary `.wasm` and would that help with 4k limit ?
Should we rather implement this in C ? 
See wabt [binary-writer.cc](https://github.com/WebAssembly/wabt/blob/main/src/binary-writer.cc)

Reply:
I think the performance would be better in C, but we would have more interop overhead. In general there is room to optimize it but right now the amount of time actually spent in it during app execution is low, so I would prefer to stay in TS until it becomes a serious performance problem.